src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -6018,16 +6020,20 @@ struct MulSimplify
   LogicalResult matchAndRewriteImpl(stablehlo::MulOp op,
                                     PatternRewriter &rewriter) const {
 
-    // 0 * x -> x
+    // 0 * x -> 0
```

Comment:
incidentally do we have this inside of nonan right now

Reply:
yeah let me update this